src/frontend/data/stores/useParticipantsStore/index.ts

```diff
@@ -0,0 +1,50 @@
+import create from 'zustand';
+
+export type ParticipantType = {
+  isInstructor: boolean;
+  isOnStage: boolean;
+  name: string;
+};
+
+type State = {
+  addParticipant: (participant: ParticipantType) => void;
+  participants: ParticipantType[];
+  removeParticipant: (participantName: string) => void;
+  updateParticipant: (participant: ParticipantType) => void;
+};
+
+export const useParticipantsStore = create<State>((set) => ({
+  participants: [],
+  addParticipant: (participant) =>
+    set((state) => {
+      if (!state.participants.some((p) => p.name === participant.name)) {
+        return {
+          participants: state.participants
+            .concat({
+              isInstructor: participant.isInstructor,
+              isOnStage: participant.isOnStage,
+              name: participant.name,
+            })
+            .sort((a, b) => a.name.localeCompare(b.name)),
+        };
+      } else {
+        return { participants: state.participants };
+      }
+    }),
+  removeParticipant: (participantName) =>
+    set((state) => ({
+      participants: state.participants.filter(
+        (p) => p.name !== participantName,
+      ),
+    })),
+  updateParticipant: (participant) =>
```

Comment:
I don't see in the code where you are updating the store. Where did you plan to do it. With your actual structure, is it easy to update a participant ? You will probably have to check if it exists and then update it or add it depending on the answer ?

Reply:
I finally used 2 different functions to update the onStage status of a participant. One for going on stage and one for going off stage. Each method only needs the name of the participant to update his status (the name is unique). Moreover if the name passed in parameter does not exist, there is no error and the code continues to run normally.